source/fe/mapping_q_generic.cc

```diff
@@ -865,20 +874,20 @@ namespace internal
                 // point is simply ignored in codimension one case. When this
                 // component is not zero, then we are projecting the point to
                 // the surface or curve identified by the cell.
-                Point<dim> p_unit_trial = p_unit;
+                Point<dim, Number> p_unit_trial = p_unit;
                 for (unsigned int i = 0; i < dim; ++i)
                   p_unit_trial[i] -= step_length * delta[i];
 
-                // shape values and derivatives
-                // at new p_unit point
-                const auto p_real_trial =
+                // shape values and derivatives at new p_unit point
+                auto p_real_trial =
```

Comment:
Why is this not const anymore?

Reply:
Effect from switching between branches. I will fix.